src/daft-dsl/src/functions/agg/mod.rs

```diff
@@ -53,7 +53,7 @@ impl ScalarUDF for MergeMeanFunction {
         }
     }
 
-    fn to_field(&self, inputs: &[ExprRef], schema: &Schema) -> DaftResult<Field> {
+    fn to_field_deprecated(&self, inputs: &[ExprRef], schema: &Schema) -> DaftResult<Field> {
```

Comment:
thoughts on marking the trait method declaration as deprecated 

https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute